### PR TITLE
Add object interceptor API to allow missing model and missing my-items handling

### LINKED ISSUE #3367: Object API is returning undefined for missing objects
Previously, missing objects would be returned as a [`MissingModel`](https://github.com/nasa/openmct/blob/14114781fcd436674fdc30d585066249b604d03f/platform/core/src/models/MissingModelDecorator.js#L40), meaning that client code could handle them like regular objects (albeit with a special type of `unknown`).

The new API does not explicitly do this, and was instead depending upon legacy code to do it. As we replace our legacy object providers this is starting to become a problem, because they do not invoke any of the legacy API that previously handled this. We need to add some catchall code to [`ObjectAPI.get()`](https://github.com/nasa/openmct/blob/14114781fcd436674fdc30d585066249b604d03f/src/api/objects/ObjectAPI.js#L156) that inspects objects returned by the `.get()` function call, and if it's undefined return a missing object. The missing object should look like this:
``` JavaScript
let missingObject = {
    identifier: requestedIdentifier,
    type: 'unknown',
    name: 'Missing: ' + openmct.objects.makeKeyString(requestedIdentified)
}
```

### LINKED ISSUE #3417: [Persistence] Couch DB provider is not working
After enabling the Couch DB persistence provided as commented [in this issue](https://github.com/nasa/openmct/issues/3201) using the default index.html, several errors appear in the console and the tree never loads.
The database exists and is public, and CORS is enabled for all domains.

Screenshot_20201005_095103

### PR DESCRIPTION
**Pull request by shefalijoshi**
Extends Object API to allow adding interceptors
ObjectAPI.get now applies any interceptors applicable to that object before returning it to the caller.

Missing model interceptor and my-items interceptors added to handle missing domain objects or missing MyItems folder.

Resolves #3417 
Resolves #3367 

## Author Checklist
Changes address original issue? Yes
Unit tests included and/or updated with changes? Yes
Command line build passes? Yes
Changes have been smoke-tested? Yes
Testing instructions included? Yes

### CODE REVIEW

File: src/api/objects/InterceptorRegistry.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    /**
+     * Register a new object interceptor.
+     *
+     * @param {string} key a string identifier for this interceptor
+     * @param {module:openmct.Interceptor} interceptor the interceptor to add
+     * @method addInterceptor
+     * @memberof module:openmct.InterceptorRegistry#
+     */
+    addInterceptor(key, interceptorDef) {
```
Comment (akhenry) [inline src/api/objects/InterceptorRegistry.js:49]: I don't *think* interceptors need keys? We don't seem to be using them here, and I can't think of a use case, but I might be missing something.

File: src/plugins/interceptors/missingObjectInterceptor.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+export default function MissingObjectInterceptor(openmct) {
+    openmct.objects.addInterceptor('missing-object', {
+        appliesTo: (identifier, domainObject) => {
```
Comment (akhenry) [inline src/plugins/interceptors/missingObjectInterceptor.js:25]: This is great, I love the separation of concerns here with the separate `appliesTo` function.

File: src/api/objects/ObjectAPI.js [hunk 3]
```diff
@@ -312,6 +319,35 @@ define([
...
 
+    /**
+     * Register a new object interceptor.
+     *
+     * @param {string} key a string identifier for this interceptor
+     * @param {module:openmct.Interceptor} interceptor the interceptor to add
+     * @method addInterceptor
+     * @memberof module:openmct.InterceptorRegistry#
+     */
+    ObjectAPI.prototype.addInterceptor = function (key, interceptorDef) {
```
Comment (akhenry) [inline src/api/objects/ObjectAPI.js:330]: Let's name this `addGetInterceptor`. We might also want to add `save` interceptors later for eg.

File: src/api/objects/ObjectAPI.js [hunk 4]
```diff
@@ -312,6 +319,35 @@ define([
         });
     };
 
+    /**
+     * Register a new object interceptor.
```
Comment (akhenry) [inline src/api/objects/ObjectAPI.js:323]: Let's add a little more documentation here to explain what get interceptors are for.

**Review by akhenry — CHANGES_REQUESTED** [review]
This looks really awesome. I'm excited about interceptors! 

I'm a fan of keeping things as simple as possible, and I don't *think* we need keys? Keys impose a requirement on developers that they come up with a unique string, and guaranteeing that a given key is unique is somewhat onerous.

File: src/api/objects/ObjectAPI.js (hunk 3 above)
Comment (shefalijoshi) [reply src/api/objects/ObjectAPI.js:330]: How do envision separating the get and save interceptors? Would they just be different objects on this class? That would work and then we wouldn't need keys either.

File: src/api/objects/InterceptorRegistry.js (hunk 1 above)
Comment (shefalijoshi) [reply src/api/objects/InterceptorRegistry.js]: Done

File: src/api/objects/ObjectAPI.js (hunk 3 above)
Comment (shefalijoshi) [reply src/api/objects/ObjectAPI.js]: Done

File: src/api/objects/ObjectAPI.js (hunk 4 above)
Comment (shefalijoshi) [reply src/api/objects/ObjectAPI.js]: Done

File: src/api/objects/ObjectAPI.js (hunk 3 above)
Comment (akhenry) [reply src/api/objects/ObjectAPI.js]: Yeah, I guess they'd be separate methods? `addGetInterceptor`, `addSaveInterceptor`.

**Review by akhenry — APPROVED** [review]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
5. Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue) Y

**Pull request merged**